src/core/observer/watcher.js

```diff
@@ -125,13 +125,11 @@ export default class Watcher {
         dep.removeSub(this)
       }
     }
-    let tmp = this.depIds
-    this.depIds = this.newDepIds
-    this.newDepIds = tmp
-    this.newDepIds.clear()
-    tmp = this.deps
-    this.deps = this.newDeps
-    this.newDeps = tmp
+    // $FlowFixMe
+    [this.depIds, this.newDepIds] = [this.newDepIds, this.depIds]
```

Comment:
Maybe leading semicolon is better for `[]`? Since the change is packed in one line so it is harder for one to forget adding semicolon.